File: ios/GMBTModule.swift
import Foundation
import Mobile
import React

@objc(GMBTModule)
class GMBTModule: NSObject {

  // MARK: - Generate New Mnemonic Seed
  @objc
  func newWordSeed(_ resolve: @escaping RCTPromiseResolveBlock,
                   rejecter reject: @escaping RCTPromiseRejectBlock) {

    var error: NSError?
    let seed = MobileNewWordSeed(&error)

    if let error = error {
      reject("SEED_ERROR", "Failed to generate seed", error)
    } else {
      resolve(seed)
    }
  }

  // MARK: - Generate Addresses
  @objc
  func getAddresses(_ seed: String,
                    num: NSNumber,
                    resolver resolve: @escaping RCTPromiseResolveBlock,
                    rejecter reject: @escaping RCTPromiseRejectBlock) {

    var error: NSError?
    let result = MobileGetAddresses(seed, num.intValue, &error)

    if let error = error {
      reject("ADDRESS_ERROR", "Failed to generate addresses", error)
    } else {
      resolve(result)
    }
  }

  // MARK: - Prepare Transaction
  @objc
  func prepareTransaction(_ inputsBody: String,
                          outputsBody: String,
                          resolver resolve: @escaping RCTPromiseResolveBlock,
                          rejecter reject: @escaping RCTPromiseRejectBlock) {

    var error: NSError?
    let tx = MobilePrepareTransaction(inputsBody, outputsBody, &error)

    if let error = error {
      reject("TX_ERROR", "Failed to prepare transaction", error)
    } else {
      resolve(tx)
    }
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}

File: android/app/src/main/java/com/monarkagarwal/gmbtcoinwallet/GMBTModule.kt
package com.monarkagarwal.gmbtcoinwallet
import com.facebook.react.bridge.*
import mobile.Mobile

class GMBTModule(private val reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "GMBTModule"
    }

    @ReactMethod
    fun getAddresses(seed: String, num: Int, promise: Promise) {
        try {
            val result = Mobile.getAddresses(seed, num.toLong())
            promise.resolve(result)
        } catch (e: Exception) {
            promise.reject("GET_ADDRESSES_ERROR", e.message, e)
        }
    }

    @ReactMethod
    fun prepareTransaction(
        inputsBody: String,
        outputsBody: String,
        promise: Promise
    ) {
        try {
            val tx = Mobile.prepareTransaction(inputsBody, outputsBody)
            promise.resolve(tx)
        } catch (e: Exception) {
            promise.reject("PREPARE_TX_ERROR", e.message, e)
        }
    }

    @ReactMethod
    fun newWordSeed(promise: Promise) {
        try {
            val seed = Mobile.newWordSeed()
            promise.resolve(seed)
        } catch (e: Exception) {
            promise.reject("NEW_SEED_ERROR", e.message, e)
        }
    }
}